bliss/simulator/decoder.py

```diff
@@ -14,55 +14,49 @@
 class ImageDecoder(nn.Module):
     def __init__(
         self,
-        n_bands: int,
         pixel_scale: float,
-        psf_params_file: str,
         psf_slen: int,
-        sdss_bands: Tuple[int, ...],
+        sdss_fields: DictConfig,
     ) -> None:
         super().__init__()
-        assert n_bands == 1, "Only 1 band is supported"
 
-        self.n_bands = n_bands
+        self.n_bands = len(sdss_fields["bands"])
         self.pixel_scale = pixel_scale
+        self.psf_slen = psf_slen
+        self.psf_galsim = {}  # Dictionary indexed by (run, camcol, field) tuple
 
-        self.params = None  # psf params from fits file
+        sdss_dir = sdss_fields["dir"]
+        sdss_bands = sdss_fields["bands"]
 
-        assert Path(psf_params_file).suffix == ".fits"
-        assert len(sdss_bands) == n_bands
-        psf_params = self._get_fit_file_psf_params(psf_params_file, sdss_bands)
-        self.params = nn.Parameter(psf_params.clone(), requires_grad=True)
+        for field_params in sdss_fields["field_list"]:
+            run = field_params["run"]
+            camcol = field_params["camcol"]
+            fields = field_params["fields"]
 
-        self.psf_slen = psf_slen
+            for field in fields:
+                # load raw params from file
+                field_dir = f"{sdss_dir}/{run}/{camcol}/{field}"
+                filename = f"{field_dir}/psField-{run:06}-{camcol}-{field:04}.fits"
```

Comment:
This is all quite specific to SDSS, even though it's in `decoder.py`, which isn't supposed to be specific to any one survey. But I see this as an issue to predates this PR, so perhaps it's fine for now. At some point though it'd be good to introduce more abstraction so we can move any SDSS-specific code to `sdss.py`.

Reply:
Sawan and I were discussing this on Friday, and we were thinking the best way to make this more general (and cleaner!) would probably be better to give Simulator a catalog-style object that contains all of the possible backgrounds/psfs/ratios to use, instead of it having to load them itself. But that would require some more general changes (and probably some new data structures) so I think it would be better to save it for a different PR.

Reply:
Sounds good